crypto/kdf/sskdf.c

```diff
@@ -430,11 +454,16 @@ static int sskdf_derive(EVP_KDF_IMPL *impl, unsigned char *key, size_t keylen)
         int ret;
         const unsigned char *custom = NULL;
         size_t custom_len = 0;
-        int nid;
+        const char *macname;
         int default_salt_len;
 
-        nid = EVP_MAC_nid(impl->mac);
-        if (nid == EVP_MAC_HMAC) {
+        /*
+         * TODO(3.0) investigate the necessity to have all these controls.
+         * Why does KMAC require a salt length that's shorter than the MD
+         * block size?
+         */
+        macname = EVP_MAC_name(impl->mac);
+        if (strcmp(macname, "HMAC") == 0) {
```

Comment:
Magic strings..

Reply:
Ok, so what do you propose?

Reply:
#define in a header, I guess we have this all over the place..

Reply:
I was thinking more in details.  What do I call that define?  Where do I put it, `core_names.h`?

Reply:
Sounds good to me.. We can always move it later if needed.

Reply:
Do you mind if that happens in a separate PR?  We have *a lot* of places where we use raw name strings...

Reply:
No I dont mind..